packages/trie/src/trie.ts

```diff
@@ -1137,17 +1148,59 @@ export class Trie {
    * @param ops
    */
   async batch(ops: BatchDBOp[], skipKeyTransform?: boolean): Promise<void> {
-    for (const op of ops) {
+    const keyTransform =
+      skipKeyTransform === true
+        ? undefined
+        : (msg: Uint8Array) => {
+            return this.appliedKey(msg)
+          }
+    const sortedOps = orderBatch(ops, keyTransform)
+    let stack: TrieNode[] = []
+    const stackPathCache: Map<string, TrieNode> = new Map()
```

Comment:
We need to make both the ordering as well as the caching optional for the `batch()` method, and therefore provide extra configuration parameters to configure.

1. **Ordering**: Reason for ordering is that the ordering in certain cases *does* have an effect on the outcome. So if there is first a put for a key and then a delete for the same key this is obviously different then having this the other way around. Same for two puts for the same key but different values. So the whole ordering needs to be deactivated by default (also for backwards compatibility) and the user would need to activate manually.

2. **Caching**: Reason that caching would need to also get an option and also would needed to be deactivated by default is that one does not know how big a batch() operation will be (only the user knows (at best)). So this might "overload" the cache respectively there is the risk for out-of-memory situations. 

I would suggest that we add a dedicated options dictionary with its own type in `types.ts` called `BatchOpts` for this, seems worth it, and then we add boolean flags `sortOperations` (or so) and `activatePathCache` (or so) for this.

My tendency actually would be to also include `skipKeyTransform` and then be a bit hacky and change the function signature to `async batch(ops: BatchDBOp[], skipKeyTransformOrOptsDict?: boolean | BatchOpts)` so that we can then do a switch respectively type check for the parameter and either assign `skipKeyTransform` directly or use the options dict.

Then we can use the method directly with `trie.batch(ops, opts)` (or `trie.batch(ops, { sortOperations: true })`) and do not always need to channel in the `skipKeyTransform` default.

(let me know if this has serious downsides though I am overlooking. I would think this should still be backwards compatible for people)

Reply:
I did some further research into the effects of this in the context of other modules, and I agree about tweaking the options for how batch works based on the situation.

There are definitely circumstances in which it would be faster to simply iterate through the batch like we currently do, instead of sorting and caching.  I also wonder how the results would change if we used different sorting and trie walking algorithms.  If there's something useful there, it could be another option in `batch()`